service/builder/receivers_builder.go

```diff
@@ -309,3 +309,51 @@ func buildFanoutMetricConsumer(pipelines []*builtPipeline) consumer.MetricsConsu
 	}
 	return processor.NewMetricsFanOutConnector(pipelineConsumers)
 }
+
+// createTraceReceiver is a helper function that creates trace receiver based on the current receiver type
+// and type of the next consumer
+func createTraceReceiver(
+	ctx context.Context,
+	factory receiver.BaseFactory,
+	logger *zap.Logger,
+	cfg configmodels.Receiver,
+	nextConsumer consumer.BaseTraceConsumer,
+) (receiver.TraceReceiver, error) {
+	if factoryV2, ok := factory.(receiver.FactoryV2); ok {
+
+		// If both receiver and consumer are of the new type (can manipulate on internal data structure),
+		// use FactoryV2.CreateTraceReceiverV2
+		if nextConsumerV2, ok := nextConsumer.(consumer.TraceConsumerV2); ok {
+			return factoryV2.CreateTraceReceiverV2(ctx, logger, cfg, nextConsumerV2)
+		}
+
+		// If receiver is of the new type, but downstream consumer is of the old type,
+		// use internalToOCTraceConverter compatibility shim
+		traceConverter := consumer.NewInternalToOCTraceConverter(nextConsumer.(consumer.TraceConsumer))
+		return factoryV2.CreateTraceReceiverV2(ctx, logger, cfg, traceConverter)
+	}
+
+	// If both receiver and consumer are of the old type (can manipulate on OC traces only),
+	// use Factory.CreateTraceReceiver
+	if nextConsumer, ok := nextConsumer.(consumer.TraceConsumer); ok {
+		return factory.(receiver.Factory).CreateTraceReceiver(ctx, logger, cfg, nextConsumer)
+	}
+
+	// Old type receiver and a new type consumer usecase is not supported
+	return nil, errors.New("OC Traces -> internal data format translation is not supported")
+}
+
+// createTraceReceiver is a helper function that creates trace receiver based on the current receiver type
+// and type of the next consumer
+// NOTE: It's just a placeholder until internal metrics data structure -> OC metrics translator is implemented
```

Comment:
```suggestion
// TODO: It's just a placeholder until internal metrics data structure -> OC metrics translator is implemented
```
Some IDEs can easily find those.